eclair-core/src/main/scala/fr/acinq/eclair/io/PendingChannelsRateLimiter.scala

```diff
@@ -59,34 +64,59 @@ private class PendingChannelsRateLimiter(nodeParams: NodeParams, router: ActorRe
           case CountOpenChannelRequests(replyTo, publicPeers) =>
             val pendingChannels = if (publicPeers) pendingPublicNodeChannels else pendingPrivateNodeChannels
             replyTo ! pendingChannels.map(_._2.length).sum
             Behaviors.same
         }
       case None =>
-        context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelIdAssigned](c => ReplaceChannelId(c.remoteNodeId, c.temporaryChannelId, c.channelId)))
-        context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelOpened](c => RemoveChannelId(c.remoteNodeId, c.channelId)))
-        context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelClosed](c => RemoveChannelId(c.commitments.remoteNodeId, c.channelId)))
-        context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelAborted](c => RemoveChannelId(c.remoteNodeId, c.channelId)))
+        context.log.info(s"restored ${pendingPublicNodeChannels.size} public peers with pending channel opens.")
+        pendingPublicNodeChannels.foreach { p => context.log.debug(s" ${p._1} -> ${p._2}")}
+        context.log.info(s"restored ${pendingPrivateNodeChannels.size} private peers with pending channel opens.")
+        pendingPrivateNodeChannels.foreach { p => context.log.debug(s" ${p._1} -> ${p._2}")}
+        context.log.debug(s"cached commands: $cachedCommands")
+        cachedCommands.foreach(context.self ! _)
         registering(pendingPublicNodeChannels, pendingPrivateNodeChannels)
     }
   }
 
   private def registering(pendingPublicNodeChannels: Map[PublicKey, Seq[ByteVector32]], pendingPrivateNodeChannels: Map[PublicKey, Seq[ByteVector32]]): Behavior[Command] = {
+    def replaceChannel(pendingNodeChannels: Map[PublicKey, Seq[ByteVector32]], remoteNodeId: PublicKey, temporaryChannelId: ByteVector32, channelId: ByteVector32): Map[PublicKey, Seq[ByteVector32]] =
+      pendingNodeChannels.get(remoteNodeId) match {
+        case Some(channels) if channels.contains(temporaryChannelId) =>
+          context.log.debug(s"replaced pending channel $temporaryChannelId with $channelId for node $remoteNodeId")
+          pendingNodeChannels + (remoteNodeId -> (channels.filterNot(_ == temporaryChannelId) :+ channelId))
+        case _ => pendingNodeChannels
+      }
+    def removeChannel(pendingNodeChannels: Map[PublicKey, Seq[ByteVector32]], remoteNodeId: PublicKey, channelId: ByteVector32): Map[PublicKey, Seq[ByteVector32]] =
+      pendingNodeChannels.get(remoteNodeId) match {
+        case Some(pendingChannels) =>
+          context.log.debug(s"removed pending channel $channelId for node $remoteNodeId")
+          val pendingChannels1 = pendingChannels.filterNot(_ == channelId)
+          if (pendingChannels1.isEmpty) {
+            pendingNodeChannels - remoteNodeId
+          } else {
+            pendingNodeChannels + (remoteNodeId -> pendingChannels1)
+          }
+        case _ => pendingNodeChannels
+      }
```

Comment:
nit: those functions can be moved outside of the `registering` function since they are pure, it's a bit easier to read and more efficient